### DISCUSSION TITLE
Idea for spellchecking typst files?

### DISCUSSION THREAD

**Discussion by KillTheMule** [typst/typst#2423] [General]
Anyone got an idea for spell checking typst files automatically/programatically?

When I was using latex, I could use `aspell -t`, and together with some bit of `sed` (because `-t` would seemingly not understand all of latex) I could simply check this against the right dictionaries.

Aspell doesn't speak typst though, so right now, I'm kinda trying to `sed` my way through, but this turns out to be both a lot of work and potentially brittle. [Text export](https://github.com/typst/typst/pull/2417) sounds right up my alley, but this probably won't be available anytime soon.

Anybody got a better idea? I'd be grateful for any hint :)

**Comment by Enivex** [comment]
There's plaintext output in the works, which would be perfect for this

**Reply by KillTheMule** [reply]
Yeah, as I linked, but it looks quite a bit in the future.

**Comment by jamesrswift** [comment]
I imagine spellchecking is more of an IDE issue?

**Reply by KillTheMule** [reply]
Naaa it's totally awesome to spellcheck your things on CI ;)

**Comment by KillTheMule** [comment]
I guess I might use `pdftotext` after rendering a pdf with typst... I'm doing the rendering anyways as part of the test pipeline, I just need to write out the file, convert it and spellcheck it. Let's see :)
